### PR TITLE
lets try self attention.

### PR DESCRIPTION
**Pull request by frankcholula**
Not the best results given that we're restricted to train with 10 epochs. It takes a lot longer to train SA. 

New model definitions and modules:

* [`src/models/self_attention.py`](diffhunk://#diff-437290e1d18188f94e066198d8dbeda41639de2e88b5ab7570c1c992309f5a52R1-R115): Added a new self-attention module and defined `resnet18_self_attention` with self-attention blocks.
* [`src/models/squeeze_and_excite.py`](diffhunk://#diff-0a2b4eb54d1fac938d05ba79d8c69422b0cbd6041e5e8da71d427e54ed3fdd8bR1-R120): Added SEBasicBlock and SEBottleneck classes, and defined `resnet18_se`, `resnet18_se_fc512`, `resnet50_se`, and `resnet50_se_fc512` with squeeze-and-excite blocks.
* [`src/models/__init__.py`](diffhunk://#diff-e66cde51b2a2f3720f6d18b345eb48713fb84166ef5dc4fe9d88780059793024R3-R14): Updated imports and model factory to include the new self-attention and SE models. [[1]](diffhunk://#diff-e66cde51b2a2f3720f6d18b345eb48713fb84166ef5dc4fe9d88780059793024R3-R14) [[2]](diffhunk://#diff-e66cde51b2a2f3720f6d18b345eb48713fb84166ef5dc4fe9d88780059793024L21-R42)

Updates to existing model files:

* [`src/models/resnet.py`](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL5-R17): Removed SEBasicBlock and SEBottleneck classes, and their corresponding model definitions, as they have been moved to `squeeze_and_excite.py`. Added `resnet18_self_attention` to the list of available models. [[1]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL5-R17) [[2]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL68-L100) [[3]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL142-L153) [[4]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL362-L375) [[5]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL393-L408) [[6]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL456-L469) [[7]](diffhunk://#diff-32f9903dd9a14d37c5300ee56d88ce85da3b030c80d54d5b026b73976800555cL485-L499)

Updates to training scripts:

* [`experiments/extra/train_resnet18_se.sh`](diffhunk://#diff-ad5c9514ac4f13e375aeec06326941337949e6ca38d9e8d08ddff7d9fbb95867L10-R11): Modified learning rate and maximum epoch parameters.
* [`experiments/extra/train_resnet18_self_attention.sh`](diffhunk://#diff-8f8318d02af33a1aba39e7941dcbc9b9ab50e523ca2ebd68739dff0c3b4606ceR1-R15): Added a new script to train ResNet18 with self-attention.

### CODE REVIEW

File: src/models/self_attention.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    Self attention module for ResNet
+
+    Implementation inspired by Non-Local Neural Networks (Wang et al.)
+    https://arxiv.org/abs/1711.07971
+    """
+
+    def __init__(self, in_channels, reduction_ratio=16):
+        super(SelfAttention, self).__init__()
+
+        # Reduce channels for computation efficiency
```
[the remark this answers is not in the document]
Comment (frankcholula) [reply src/models/self_attention.py:18]: bad bot
